File: jv-service/src/main/java/com/lcaohoanq/jvservice/domain/chat/ChatRoomRepository.java
package com.lcaohoanq.jvservice.domain.chat;

import com.lcaohoanq.jvservice.domain.user.User;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ChatRoomRepository extends JpaRepository<ChatRoom, Long> {
    
    Optional<ChatRoom> findByUser1AndUser2(User user1, User user2);

}

File: chat-service/src/main/kotlin/com/lcaohoanq/chatservice/domain/chatroom/ChatRoomRepository.kt
package com.lcaohoanq.chatservice.domain.chatroom

import org.springframework.data.jpa.repository.JpaRepository

interface ChatRoomRepository : JpaRepository<ChatRoom, String> {
    fun findBySenderIdAndRecipientId(senderId: String, recipientId: String): ChatRoom?
}
